tests/test_segment_court.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+"""
+Integration Test: segment_court_frames
+
+This script tests the `segment_court_frames` function from `sam_utils`, which performs segmentation 
+on the basketball court using the SAM2 video segmentation model.
+
+The function extracts the court shape from the initial video frame and propagates that mask through the 
+entire video, generating overlays to visualize court segmentation across frames. The resulting video frames 
+with overlays are displayed in a GUI window and optionally saved to an output file if `utils.out` is initialized.
+
+Usage:
+    Run the script directly to visualize and optionally save the court-segmented video.
+
+Requirements:
+    - The `HF_MODEL` checkpoint path must be defined in `config.py`.
+    - The input video should exist at `videos/louisville_60s_clip.mp4`.
+"""
+
+import os
+import cv2
+import sys
+from pathlib import Path
+sys.path.append(str(Path(__file__).resolve().parents[1]))
+from experimental.segmentation import segment_court_frames
+from config import HF_MODEL, TMP_DIRNAME_VIDEOS
+
+def test_segment_court():
+    assert not os.path.exists(os.path.join(TMP_DIRNAME_VIDEOS, "louisville_60s_clip.mp4"))
+    video_path = os.path.join(os.getcwd(), "videos/louisville_60s_clip.mp4")
+    for overlay in segment_court_frames(video_path, HF_MODEL):
+        cv2.imshow(__file__, overlay)
+        
+    assert os.path.exists(os.path.join(TMP_DIRNAME_VIDEOS, "louisville_60s_clip.mp4"))
```

Comment:
@gg-blake I think the condition here might need to be changed (somehow) to get the test to pass

Reply:
So we spoke about this in a thread on the discord server, let's take the tests out from here and get the code merged. We can do a separate PR that moves the tests to the lower level funtions in the PR. 

Thanks @gg-blake